src/dune_rules/dialect.ml

```diff
@@ -208,18 +216,18 @@ module DB = struct
   let set_extensions_for_merlin t =
     let v =
       fold t ~init:[] ~f:(fun d s ->
-          let impl = extension d Ml_kind.Impl in
+          let impl = Option.value_exn @@ extension d Ml_kind.Impl in
           let intf = extension d Ml_kind.Intf in
           if
             (* Only include dialects with no preprocessing and skip default file
                extensions *)
             preprocess d Ml_kind.Impl <> None
             || preprocess d Ml_kind.Intf <> None
-            || impl = extension ocaml Ml_kind.Impl
+            || impl = Option.value_exn @@ extension ocaml Ml_kind.Impl
```

Comment:
same question here, why can't this raise?

Reply:
done